orderer/consensus/etcdraft/util.go

```diff
@@ -297,10 +299,30 @@ func (conCert ConsenterCertificate) IsConsenterOfChannel(configBlock *common.Blo
 		return err
 	}
 
+	bl, _ := pem.Decode(conCert.ConsenterCertificate)
+	if bl == nil {
+		return errors.Errorf("my consenter certificate %s is not a valid PEM", string(conCert.ConsenterCertificate))
+	}
+
+	myCertDER := bl.Bytes
+
 	for _, consenter := range m.Consenters {
-		if bytes.Equal(conCert.ConsenterCertificate, consenter.ServerTlsCert) || bytes.Equal(conCert.ConsenterCertificate, consenter.ClientTlsCert) {
+		candidateBlock, _ := pem.Decode(consenter.ServerTlsCert)
+		if candidateBlock == nil {
+			return errors.Errorf("candidate server certificate %s is not a valid PEM", string(consenter.ServerTlsCert))
+		}
+		sameServerCertErr := crypto.CertificatesWithSamePublicKey(myCertDER, candidateBlock.Bytes)
+
+		candidateBlock, _ = pem.Decode(consenter.ClientTlsCert)
+		if candidateBlock == nil {
+			return errors.Errorf("candidate client certificate %s is not a valid PEM", string(consenter.ClientTlsCert))
+		}
+		sameClientCertErr := crypto.CertificatesWithSamePublicKey(myCertDER, candidateBlock.Bytes)
+
+		if sameServerCertErr == nil || sameClientCertErr == nil {
 			return nil
 		}
+		conCert.Logger.Debugf("I am not %s:%d because %s, %s", consenter.Host, consenter.Port, sameServerCertErr, sameClientCertErr)
```

Comment:
If I read this correctly, because you've made the optimization around deferring the error text computation, when in this log message, these errors will no longer have the public key bytes rendered?

Perhaps here is not the right place... but I expect it would be helpful in debugging to ensure that we are printing the hex or base64 encoded bytes of our public key, and those of each certificate in the channel config somewhere.  Perhaps it's good enough that we are printing full PEMs in other places, which can always be used to extract the public key, but, just a thought.

Reply:
> If I read this correctly, because you've made the optimization around deferring the error text computation, when in this log message, these errors will no longer have the public key bytes rendered?

You are not reading this correctly, this is unrelated. 
I put caching in a single place - when you read the certificate from the gRPC stream as a server, you cache the invocations of the function. All other invocations are not cached:
- Client side invocations (you as a client, authenticate a server) do not require caching because you connect once and persist the TLS connection until you disconnect. 
- All replication related invocations are either due to onboarding which is slow anyway, or are when Raft catches up, which means you are behind the rest, hence not contributing to any throughput and not in the critical path.

> Perhaps here is not the right place... but I expect it would be helpful in debugging to ensure that we are printing the hex or base64 encoded bytes of our public key, and those of each certificate in the channel config somewhere. 

I added a print in case a match is not found, which is most likely when you would want to troubleshoot.